ksql-common/src/main/java/io/confluent/ksql/schema/connect/SqlSchemaFormatter.java

```diff
@@ -78,85 +68,81 @@ public SqlSchemaFormatter(final Set<String> reservedWords, final Option... optio
 
   @Override
   public String format(final Schema schema) {
-    return formatSchema(new Context(schema, options, true, reservedWords));
+    final String converted = SchemaWalker.visit(schema, new Converter()) + typePostFix(schema);
+
+    return options.contains(Option.AS_COLUMN_LIST)
+        ? stripTopLevelStruct(converted)
```

Comment:
Will there ever be a situation where we try to strip the top-level struct from a struct marked as `NOT NULL`? It doesn't look like the current implementation of `stripTopLevelStruct()` would handle that correctly.

Reply:
Good catch - fixed.